DataDog/datadog-agent
# forwarder retry queue payloads max size

### What does this PR do?

Add the config `forwarder_retry_queue_all_payloads_max_size` which defines the maximum size in bytes of all the payloads in the forwarder's retry queue. If `forwarder_retry_queue_all_payloads_max_size` is greater than 0,    `forwarder_retry_queue_max_size` is ignored.

### Motivation

`forwarder_retry_queue_max_size` does not take into account the fact that payloads have very different sizes.

### Additional Notes

`forwarder_retry_queue_all_payloads_max_size` is defined as bytes which mean the number is very big. I have tried to be consistent with other settings like `secret_backend_output_max_size`.

### Describe your test plan
- Check `forwarder_retry_queue_max_size` works as before.
- Setup a local proxy for (example`dd_url: http://127.0.0.1:1234`) that dump requests to a file but returns an HTTP error.
- Set `forwarder_retry_queue_all_payloads_max_size` to `15000000`
- When you see the error `Dropped %d transactions in this retry attempt: %d for exceeding the retry queue size limit of %d, %d because the workers are too busy` in the log, stop the proxy and make sure the sum of all payloads are greater than  `15000000`
